### ISSUE TITLE
Comparing colors with new Keyboard Grid returns false

### ISSUE THREAD

**Issue by WolfspiritM** [CoraleStudios/Colore#167]
With the new Grid implementation the returned Color from custom[Key] returns the Color Value including the set Key Flag which results in another Value then the one written into it.
That means that something like that always returns false:

``` C#
            custom[Key.Z] = Color.White;
            var colkey = custom[Key.Z];
            if (colkey == Color.White)
            {
                Console.WriteLine("Key is white.");
            }
```

**Comment by Sharparam** [comment]
We can either fix this by changing the return in `Custom`'s `Key` accessor or in `Color` itself, what do you think is the best approach @brandonscott?

**Comment by brandonscott** [comment]
The Color class contains the flag to determine the location. As such, I think a comparison should be specifically in the Key accessor for Custom. Otherwise a comparison ignoring the flag isn't a true comparison?

**Comment by Sharparam** [comment]
There are no comparisons being done in the `Key` accessor. What we can change is that `this[Key].get` in `Custom` returns a `Color` instance with the layout bit removed to facilitate comparing with other color values.

The alternative would be to, in `Color`, hide the actual value in the backing field and make the publicly exposed `Value` property always return a value that has the upper-most byte removed. This means we'll be doing modifications every time the `Value` property is accessed in some way which might not be ideal.

**Comment by brandonscott** [comment]
Sorry, I mean a modification to the accessor so that the comparisons can happen.

Yes, your first example is what I had in mind.

**Comment by brandonscott** [comment]
Who fancies implementing this one?

**Comment by Sharparam** [comment]
So we're doing it with the change in `Custom`'s `Key` accessor?

**Comment by brandonscott** [comment]
I think that would be best.

**Comment by Sharparam** [comment]
Fixed in #175
